pkg/validations/commandvalidations/commandvalidations.go

```diff
@@ -0,0 +1,82 @@
+package commandvalidations
+
+import (
+	"context"
+	"fmt"
+	"runtime"
+
+	"github.com/aws/eks-anywhere/pkg/cluster"
+	"github.com/aws/eks-anywhere/pkg/dependencies"
+	"github.com/aws/eks-anywhere/pkg/validations"
+	"github.com/aws/eks-anywhere/pkg/workflows/interfaces"
+)
+
+type ValidateCreateCluster struct {
+	clusterSpec    *cluster.Spec
+	deps           *dependencies.Dependencies
+	cmdValidations interfaces.Validator
+	dockerExec     validations.DockerExecutable
+}
+
+func New(clusterSpec *cluster.Spec, deps *dependencies.Dependencies, cmdValidations interfaces.Validator, dockerExec validations.DockerExecutable) *ValidateCreateCluster {
+	return &ValidateCreateCluster{
+		clusterSpec:    clusterSpec,
+		deps:           deps,
+		cmdValidations: cmdValidations,
+		dockerExec:     dockerExec,
+	}
+}
+
+func (v *ValidateCreateCluster) ValidateCreate(ctx context.Context) error {
+	runner := validations.NewRunner()
+	runner.Register(v.generateCreateValidations(ctx)...)
+	runner.Register(v.deps.GitOpsFlux.Validations(ctx, v.clusterSpec)...)
+	err := runner.Run()
+
+	return err
+}
+
+func (v *ValidateCreateCluster) generateCreateValidations(ctx context.Context) []validations.Validation {
+	return []validations.Validation{
+		func() *validations.ValidationResult {
+			return &validations.ValidationResult{
+				Name:   "validate docker executable",
+				Err:    ValidateDockerExecutable(ctx, v.dockerExec, runtime.GOOS),
+				Silent: true,
+			}
+		},
+		func() *validations.ValidationResult {
+			return &validations.ValidationResult{
+				Name:   "validate kubeconfig path",
+				Err:    ValidateKubeconfigPath(v.clusterSpec.Cluster.Name),
+				Silent: true,
+			}
+		},
+		func() *validations.ValidationResult {
+			return &validations.ValidationResult{
+				Name:   "validate cluster",
+				Err:    v.clusterSpec.Cluster.Validate(),
```

Comment:
should we validate the whole cluster config?

Reply:
Good catch, using cluster.ValidateConfig instead